Samples: Bluetooth: Add CAP initiator unicast sample

Thalley opened the pull request:
Adds a CAP initiator and CAP acceptor sample that only does unicast, along with matching babblesim tests for these new samples

Fixes https://github.com/zephyrproject-rtos/zephyr/issues/71607

Thalley committed:
Samples: Bluetooth: Add CAP initiator unicast sample

Adds a CAP initiator unicast sample that simply
uses the CAP initiator unicast API to setup
streams with a CAP acceptor.

To keep it simple no audio or encoding support
has been added.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Samples: Bluetooth: Add CAP acceptor unicast sample

Adds a CAP acceptor unicast sample that simply
uses the CAP acceptor implementation to setup
streams with a CAP initiator.

To keep it simple no audio or encoding support
has been added.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
tests: Bluetooth: Audio: Add BSIM test of the CAP samples

Adds babblesim tests of the CAP Acceptor and the CAP Initiator
samples for unicast.

This simply checks if at least one stream is connected
and is sending (empty) data.

This modifies the samples to send data as well as counting the
receive ISO data packets.
Ideally the TX would be superflous to verify that ISO is working,
but a missing feature in the Zephyr LL makes it a requirement.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

aescolar commented on tests/bsim/bluetooth/audio_samples/compile.sh:
```diff
@@ -18,6 +18,12 @@ if [ "${BOARD_TS}" == "nrf5340bsim_nrf5340_cpuapp" ]; then
     conf_file=${ZEPHYR_BASE}/samples/bluetooth/broadcast_audio_sink/prj.conf \
     exe_name=bs_${BOARD_TS}_${app}_prj_conf sysbuild=1 compile
+  app=tests/bsim/bluetooth/audio_samples/cap_initiator sysbuild=1 \
+    conf_file=${ZEPHYR_BASE}/samples/bluetooth/broadcast_audio_sink/prj.conf \
```
is this a copy paste error or you really want different conf files for the nrf52 and 53?

Thalley replied:
Spelling mistake :)

kartben commented on samples/bluetooth/cap_acceptor/README.rst:
```diff
@@ -0,0 +1,74 @@
+********
+
+Application demonstrating the CAP Acceptor functionality.
+Starts by advertising for a CAP Initiator to connect set up available streams.
```
incomplete sentence

kartben commented on samples/bluetooth/cap_acceptor/README.rst:
```diff
@@ -0,0 +1,74 @@
+.. zephyr:code-sample:: bluetooth_cap_acceptor
+   :name: Bluetooth: Common Audio Profile Acceptor
+   :relevant-api: bluetooth
```
```suggestion
   :relevant-api: bt_cap bluetooth
```

kartben commented on samples/bluetooth/cap_initiator/README.rst:
```diff
@@ -0,0 +1,74 @@
+.. zephyr:code-sample:: bluetooth_cap_initiator
+   :name: Bluetooth: Common Audio Profile Initiator
+   :relevant-api: bluetooth
```
```suggestion
   :relevant-api: bt_cap bluetooth
```

kartben reviewed (dismissed):
some doc-related comments

Thalley committed:
MAINTAINERS: Add BT CAP samples to LE audio group

Add the CAP initiator and acceptor samples to the files
for Bluetooth Audio.

This will also automatically support a future
CAP Commander sample.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

kartben commented on samples/bluetooth/cap_acceptor/README.rst:
```diff
@@ -0,0 +1,74 @@
+   :name: Bluetooth: Common Audio Profile Acceptor
+   :relevant-api: bt_cap bt_bap bluetooth
+
+   Bluetooth: Common Audio Profile Acceptor
```
Just noticed this one: this should actually be a short description of what the sample is about. It is recommended to word this as if you were completing the sentence "This code sample shows how to ...". Otherwise https://builds.zephyrproject.io/zephyr/pr/72075/docs/connectivity/bluetooth/api/audio/cap.html is not as useful as it could be :)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/a34b4ed4-1e01-48c4-9525-0e0d06521b05)


So maybe: 

```suggestion
   Receive audio data from a Bluetooth device using the Common Audio Profile (CAP) Acceptor role
```

don't know if you maybe also want to have "unicast" as a keyword in that description as it looks like the sample might do multicast in the future?

Thalley replied:
Yeah, it will support broadcast soon as well.

kartben commented on samples/bluetooth/cap_initiator/README.rst:
```diff
@@ -0,0 +1,74 @@
+   :name: Bluetooth: Common Audio Profile Initiator
+   :relevant-api: bt_cap bt_bap bluetooth
+
+   Bluetooth: Common Audio Profile Initiator
```
```suggestion
   Send audio data to a Bluetooth device using the Common Audio Profile (CAP) Initiator role
```

kartben replied:
@Thalley  was there a reason not to use the suggested wording? For the sake of consistency with other samples I deliberately proposed a wording aligned with the recommendations in https://github.com/zephyrproject-rtos/zephyr/blob/main/doc/templates/sample.tmpl

Thalley replied:
It doesn't send audio data 8) It's just dummy data that cannot be played back

kartben reviewed (commented):
another small comment re: documentation - thanks (and sorry I missed it earlier!)

larsgk reviewed (changes requested):
Great work

The code looks good to me, however, I think it would make sense to include some comments/descriptions on the functions in the samples to better understand the flow for newcomers.

Also, I would like to run the samples on nrf5340dk and ADK before approving

Thalley commented:
> Great work
> 
> The code looks good to me, however, I think it would make sense to include some comments/descriptions on the functions in the samples to better understand the flow for newcomers.
> 
> Also, I would like to run the samples on nrf5340dk and ADK before approving

Added additional comments and documentation. Please feel free to point out more places where it can be improved

Thalley commented:
Rebased to solve merge conflicts

Thalley commented:
The compile issues seem to be related to x86 targets only. Investigating what the cause is, but shouldn't affect the overall PR, so it can still be reviewed

Thalley commented:
CI is failing on non-ARM builds. Should be fixed by https://github.com/zephyrproject-rtos/zephyr/pull/73768

Thalley commented:
@kartben Please re-review or dismiss change requests
